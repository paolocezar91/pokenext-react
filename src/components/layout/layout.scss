main {
    height: calc(100vh - 82px - 34px);
    background-color: var(--background);
}