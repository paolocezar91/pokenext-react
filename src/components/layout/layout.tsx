import "@/app/globals.css";
import Footer from "@/components/layout/footer";
import "./layout.scss";
import Navbar from "@/components/layout/navbar/navbar";
import Head from "next/head";

export default function RootLayout({
  children,
  title,
}: Readonly<{
  children: React.ReactNode;
  title: string;
}>) {

  title = 'Pok√©Next -- ' + title;

  return (
    <>
      <Head>
        <title>{title}</title>
      </Head>
      <div className="container mx-auto">
        <Navbar title={title}/>
        <main className="sm:w-min md:w-[initial]">
          {children}
        </main>
        <Footer />
      </div>
    </>
  );
}
