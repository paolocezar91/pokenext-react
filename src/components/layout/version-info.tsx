import { buildInfo } from '@/build-info';

export function VersionInfo() {

  return (
    <div className="text-xs text-white">
      <div>v{buildInfo.version}</div>
    </div>
  );
}