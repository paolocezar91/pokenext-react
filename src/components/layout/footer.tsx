import LangSelect from "../settings/lang-select";
import { VersionInfo } from "./version-info";

export default function Footer() {
  return (
    <div
      className="
        footer
        container
        w-full
        fixed
        bottom-0
        flex
        justify-between
        border-solid
        border-t-2
        border-(--pokedex-red-light)
        bg-(--pokedex-red)
        py-2
        px-4
      "
    >
      <div className="w-full flex items-center justify-between text-xs text-white">
        <VersionInfo />
        <LangSelect />
      </div>
    </div>
  );
}
