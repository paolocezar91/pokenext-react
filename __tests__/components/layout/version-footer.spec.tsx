import Footer from "@/components/layout/footer";
import { VersionInfo } from "@/components/layout/version-info";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";

describe("VersionInfo", () => {
  it("should match snapshot with no valid description", () => {
    const { asFragment } = render(<VersionInfo />);
    expect(asFragment()).toMatchSnapshot();
  });
});
