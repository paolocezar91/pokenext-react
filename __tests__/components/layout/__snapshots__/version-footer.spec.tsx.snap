// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VersionInfo should match snapshot with no valid description 1`] = `
<DocumentFragment>
  <div
    class="text-xs text-white"
  >
    <div>
      v0.4.1
    </div>
  </div>
</DocumentFragment>
`;
